src/handlers/api.js

```diff
@@ -189,47 +189,123 @@ export const apiGetGasPrices = () => api.get(`/get_eth_gas_prices`);
  */
 export const apiShapeshiftGetCurrencies = () => api.get(`/get_currencies`);
 
+/**
+ * Configuration for shapeshift api
+ * @type axios instance
+ */
+const shapeshift = axios.create({
+  baseURL: 'https://shapeshift.io',
+  timeout: 30000, // 30 secs
+  headers: {
+    'Content-Type': 'application/json',
+    Accept: 'application/json',
+  },
+});
+
 /**
  * @desc shapeshift get market info
- * @param  {String}   [depositSelected = '']
- * @param  {String}   [withdrawalSelected = '']
+ * @param  {String}   [pair = '']
  * @return {Promise}
  */
-export const apiShapeshiftGetMarketInfo = (
-  depositSelected = '',
-  withdrawalSelected = '',
-) =>
-  api.get(
-    `/get_market_info?deposit=${depositSelected}&withdrawal=${withdrawalSelected}`,
-  );
+export const apiShapeshiftGetMarketInfo = (pair = '') =>
+  shapeshift.get(`/marketinfo/${pair}`);
 
 /**
- * @desc shapeshift get fixed price
- * @param  {String}   [amount = '']
- * @param  {String}   [exchangePair = '']
- * @param  {String}   [address = '']
+ * @desc shapeshift get fixed or quoted price
+ * @param  {String}   [depositSymbol = '']
+ * @param  {String}   [withdrawalSymbol = '']
+ * @param  {String}   [depositAmount = '']
+ * @param  {String}   [withdrawalAmount = '']
  * @return {Promise}
  */
-export const apiShapeshiftGetFixedPrice = (
-  amount = '',
-  exchangePair = '',
+export const apiShapeshiftSendAmount = async ({
   address = '',
-) =>
-  api.post(`/shapeshift_send_amount`, {
-    amount,
-    withdrawal: address,
-    pair: exchangePair,
-    returnAddress: address,
-  });
+  depositSymbol = '',
+  withdrawalSymbol = '',
+  depositAmount = '',
+  withdrawalAmount = '',
+}) => {
+  try {
+    const pair = `${depositSymbol.toLowerCase()}_${withdrawalSymbol.toLowerCase()}`;
+    const marketInfo = await apiShapeshiftGetMarketInfo(pair);
+    const min = marketInfo.data.minimum;
+    const body = address
+      ? {
+          pair,
+          withdrawal: address,
+          returnAddress: address,
+        }
+      : { pair };
+    if (withdrawalAmount) {
+      body.amount = withdrawalAmount;
+    } else if (depositAmount) {
+      body.depositAmount = depositAmount;
+    } else {
+      body.depositAmount = min;
+    }
+    const shapeshiftApiKey = process.env.REACT_APP_SHAPESHIFT_API_KEY || '';
+    if (shapeshiftApiKey) {
+      body.apiKey = shapeshiftApiKey;
+    }
+    console.log('shapeshift sendamount body', body);
```

Comment:
Remove `console.log`?

Reply:
This was some testing that @jinchung was working on for the Shapeshift API Key.

Thanks anyway, I will label this PR as WIP